### PR TITLE
Update h1,h2,h3 font sizes in the docs pages

### LINKED ISSUE #2936: H1 header is smaller than H2 in docs pages -- broken CSS?
For some reason we are seeing H1 on pages in the docs looking smaller than H2 (see https://docs.kedro.org/en/stable/tutorial/spaceflights_tutorial.html for an example). I don't know why this is or when it started but it seems OK in Kedro 0.18.0 https://docs.kedro.org/en/0.18.0/tutorial/spaceflights_tutorial.html

Would it be possible to investigate further please?

Edit to note: Also H3 looks to be only marginally smaller than H2.

I should also note that we are likely to make some more far-reaching changes to the docs design in the near-ish future so this shouldn't turn into a major revision of the CSS (however ugly it is) and if it looks like a big chunk of work, we can live with it for now!

### PR DESCRIPTION
**Pull request by tynandebold**
## Description

Resolves #2936.

## Development notes

Small changes to CSS.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Reviewing when the RTD builds work again 😢

**Comment by stichbury** [comment]
@tynandebold I cherry-picked your commit into my branch (which I'm building and publishing to https://stichbury.github.io/index.html so that it's easy to review my PR #2888). So the changes are now visible over on that build...and they're great. Thank you!

image

**Review by stichbury — APPROVED** [review]
🌟

File: docs/source/_static/css/qb1-sphinx-rtd.css [hunk 1]
```diff
@@ -321,16 +321,16 @@ h1, h2, .rst-content .toctree-wrapper p.caption, h3, h4, h5, h6, legend {
...
 }
 
 .wy-body-for-nav h2 {
-    font-size: 2.3rem;
+    font-size: 2rem;
 }
 
 .wy-body-for-nav h3 {
-    font-size: 2.1rem;
+    font-size: 2rem;
```
Comment (astrojuanlu) [inline docs/source/_static/css/qb1-sphinx-rtd.css:333]: Haven't checked how this looks like yet, but are we making h2 and h3 the same size then?

Comment (astrojuanlu) [reply docs/source/_static/css/qb1-sphinx-rtd.css:333]: Just got around and saw it in @stichbury local build https://stichbury.github.io/data/data_catalog.html#additional-settings-in-catalog-yml I'm wondering if we could indeed make h3 smaller. I see h4 are already quite big as well, maybe what about h3 1.8rem, h4 1.7rem?

Comment (tynandebold) [reply docs/source/_static/css/qb1-sphinx-rtd.css:333]: I made a mistake. h2 should be 2.3 still. Let me change.

I don't ever see any h4's on the page. I don't think we need to change that one.

**Review by astrojuanlu — APPROVED** [review]

**Pull request merged**